provider/confluence/api.ts

```diff
@@ -0,0 +1,70 @@
+import type { Settings } from './index.ts'
+
+const authHeaders = (settings: Settings) => ({
+    Authorization: `Basic ${Buffer.from(`${settings.email}:${settings.apiToken}`).toString('base64')}`,
+})
+
+const endpoint = (settings: Settings) =>
+    'https://' + settings.host + (settings.port ? `:${settings.port}` : '')
+
+export interface Page {
+    title: string
+    body: string
+    space: {
+        name: string
+    }
+    uri: string
+}
+
+interface PagesSearchJSON {
+    results: {
+        title: string
+        space: {
+            name: string
+        }
+        body: {
+            storage: {
+                value: string
+            }
+        }
+        _links: {
+            webui: string
+        }
+    }[]
+}
+
+const escapeCqlQuotes = (input: string) => input.replace(/"/g, '\\"')
```

Comment:
minor: would a more robust version of this function be `JSON.stringify` and then at the call site you don't wrap it in `"`. Maybe then name it `quoteCql`?